main.tf

```
@@ -185,10 +185,12 @@ module "datadog_integration_role" {
   name          = local.datadog_integration_role_name
   assume_policy = data.aws_iam_policy_document.datadog_integration_assume_role.json
   create_policy = true
-  policy_arns   = var.cspm_resource_collection_enabled ? ["arn:aws:iam::aws:policy/SecurityAudit", "arn:aws:iam::${data.aws_caller_identity.current.account_id}:policy/${local.datadog_resource_collection_policy_name}"] : []
+  policy_arns   = var.extended_resource_collection_enabled ? ["arn:aws:iam::aws:policy/SecurityAudit", "arn:aws:iam::${data.aws_caller_identity.current.account_id}:policy/${local.datadog_resource_collection_policy_name}"] : []
```

```suggestion
  policy_arns   = var.extended_resource_collection_enabled ? ["arn:aws:iam::aws:policy/SecurityAudit", aws_iam_policy.datadog_resource_collection_policy.arn] : []
```

And this doesn't work because it'll complain about wanting to use target mode cause it can't compute if to create the resource when running for the first time?

---

yes it broke things like that. I tried